### PR TITLE
Multiple misc changes to shader interfaces

### PR DESCRIPTION
**Pull request by IMS212**

### CODE REVIEW

File: common/src/main/java/net/caffeinemc/mods/sodium/client/gl/shader/GlProgram.java [hunk 1]
```diff
@@ -51,22 +51,30 @@ public void delete() {
     }
 
     @Override
-    public <U extends GlUniform<?>> U bindUniform(String name, IntFunction<U> factory) {
+    public <U extends GlUniform<?>> U bindUniform(String name, IntFunction<U> factory, boolean required) {
```
Comment (jellysquid3) [inline common/src/main/java/net/caffeinemc/mods/sodium/client/gl/shader/GlProgram.java]: I'd rather see a `bindUniformOptional` which returns `@Nullable`, and a `bindUniform` which returns `@NotNull` and does the check itself.

File: common/src/main/java/net/caffeinemc/mods/sodium/client/gl/shader/GlProgram.java [hunk 2]
```diff
@@ -51,22 +51,30 @@ public void delete() {
...
+                return null;
+            }
         }
 
         return factory.apply(index);
     }
 
     @Override
-    public GlUniformBlock bindUniformBlock(String name, int bindingPoint) {
+    public GlUniformBlock bindUniformBlock(String name, int bindingPoint, boolean required) {
```
Comment (jellysquid3) [inline common/src/main/java/net/caffeinemc/mods/sodium/client/gl/shader/GlProgram.java]: See comment about `bindUniform`.

File: common/src/main/java/net/caffeinemc/mods/sodium/client/render/vertex/serializers/VertexSerializerRegistryImpl.java [hunk 3]
```diff
@@ -46,6 +46,11 @@ public VertexSerializer get(VertexFormatDescription srcFormat, VertexFormatDescr
         return serializer;
     }
 
+    @Override
+    public void registerSerializer(VertexFormatDescription srcFormat, VertexFormatDescription dstFormat, VertexSerializer serializer) {
```
Comment (jellysquid3) [inline common/src/main/java/net/caffeinemc/mods/sodium/client/render/vertex/serializers/VertexSerializerRegistryImpl.java:50]: I'm concerned about race conditions if Iris doesn't call into this early enough. But if it's handled in the client initializer, it's _probably_ fine?

**Comment by IMS212** [comment]
Fixed the two things; not sure if any changes should be made to the serializer portion, or if that can wait on b2.

**Review by jellysquid3 — APPROVED** [review]

**Pull request merged**